velox/substrait/SubstraitToVeloxPlan.h

```diff
@@ -0,0 +1,134 @@
+/*
+ * Copyright (c) Facebook, Inc. and its affiliates.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#pragma once
+
+#include "SubstraitToVeloxExpr.h"
+#include "velox/connectors/hive/HiveConnector.h"
+
+namespace facebook::velox::substrait {
+
+/// This class is used to convert the Substrait plan into Velox plan.
+class SubstraitVeloxPlanConverter {
+ public:
+  /// Used to convert Substrait AggregateRel into Velox PlanNode.
+  std::shared_ptr<const core::PlanNode> toVeloxPlan(
+      const ::substrait::AggregateRel& sAgg);
+
+  /// Used to convert Substrait ProjectRel into Velox PlanNode.
+  std::shared_ptr<const core::PlanNode> toVeloxPlan(
+      const ::substrait::ProjectRel& sProject);
+
+  /// Used to convert Substrait FilterRel into Velox PlanNode.
+  std::shared_ptr<const core::PlanNode> toVeloxPlan(
+      const ::substrait::FilterRel& sFilter);
+
+  /// Used to convert Substrait ReadRel into Velox PlanNode.
+  std::shared_ptr<const core::PlanNode> toVeloxPlan(
+      const ::substrait::ReadRel& sRead,
+      u_int32_t& index,
+      std::vector<std::string>& paths,
+      std::vector<u_int64_t>& starts,
+      std::vector<u_int64_t>& lengths);
+
+  /// Used to convert Substrait Rel into Velox PlanNode.
+  std::shared_ptr<const core::PlanNode> toVeloxPlan(
+      const ::substrait::Rel& sRel);
+
+  /// Used to convert Substrait RelRoot into Velox PlanNode.
+  std::shared_ptr<const core::PlanNode> toVeloxPlan(
+      const ::substrait::RelRoot& sRoot);
+
+  /// Used to convert Substrait Plan into Velox PlanNode.
+  std::shared_ptr<const core::PlanNode> toVeloxPlan(
+      const ::substrait::Plan& sPlan);
+
+  /// Will return the index of Partition to be scanned.
+  u_int32_t getPartitionIndex() {
+    return partitionIndex_;
+  }
+
+  /// Will add the paths of the files to be scanned into a vector.
+  void getPaths(std::vector<std::string>& paths) {
```

Comment:
perhaps a more idiomatic way to achieve this is to have a function exposing the path_ vector as a const ref, and if you need to reuse this pattern inside the .cpp, this could be a free function inside an anonymous namespace on that file?

Reply:
Thanks for your advice.
Actually these methods are called outside the .cpp, and in this PR, they are called in the UT. Changed them to expose the vector as a const ref.